### PR TITLE
Add tags to Response List

### PR DESCRIPTION
**Pull request by greeffjean**
Please review, tag filtering functionality added, Tag API service created. List UI updated

### CODE REVIEW

**Comment by avishkar58** [comment]
Hi Jean,
Thanks for this, I'll review it after your previous PRs have been merged so theres a clean diff

**Comment by greeffjean** [comment]
Hi Avishkar.

Thank you for the payment, did you have a little break / holiday?

Can you please assign me a new issue in the mean time?

Have a great day!
Kind Regards

Jean Greeff

**Comment by avishkar58** [comment]
Hi @greeffjean please merge develop in here and resolve any conflicts then I can review :)

**Review by avishkar58 — CHANGES_REQUESTED** [review]
Hi Jean,

Great work! I really like what you did with the tag selection UI! Two suggestions to improve the UI a bit:
1. I initially found the Update button on the tag selection dialog confusing, because its positioning and shape is the same as the tags, so I initially thought it was a selected tag or something. It would be clearer if the button was positioned differently or more prominent like you have it on the questions dialog
2. The reset button on the questions dialog resets both the questions and the tags - so it should probably be outside the question dialog, for example, on the same row as the Tags and Questions button, aligned to the right

Thanks!

**Comment by avishkar58** [comment]
Hey Jean, maybe I was imagining, but didn't you previously have the ability to click on the reponse_id and it takes you to the responsePage? That seems to have disappeared?

File: webapp/src/pages/ResponseList/components/ResponseListForm.js [hunk 1]
```diff
@@ -1,76 +1,71 @@
...
-        this.handleData();
+        this.fetchTags()
     }
 
-
-    fetchData() {
-        fetchQuestions().then(response => {
+    // Fetch Tags
+    fetchTags() {
+        tagList.list().then(response => {
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponseList/components/ResponseListForm.js:32]: Oh yes, again here I made an update in a later issue, made a merge but I see the **tagList.service.js** was not updated in accordance.

File: webapp/src/pages/ResponseList/components/ResponseListForm.js [hunk 2]
```diff
@@ -163,28 +165,73 @@ class ResponseListForm extends Component {
...
-            }
-            )
+                btnUpdate: false,
+            })
+        })
+    }
+
+    // Tag Selection State
+    tagSelector(name) {
+        let list = this.state.selectedTags;
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponseList/components/ResponseListForm.js:189]: Thanks!

**Comment by greeffjean** [comment]
> Hey Jean, maybe I was imagining, but didn't you previously have the ability to click on the reponse_id and it takes you to the responsePage? That seems to have disappeared?

` // Create Response Id Link
                if (this.props.event) {
                    val.response_id = <NavLink
                        to={`${this.props.event.key}/responsePage/${val.response_id}`}
                        className="table-nav-link"
                    >
                        {val.response_id}
                    </NavLink>;
                }`

Whats suppose to be happening here is the link you saw before only renders if the there is an event prop present in this component, since we need to write this line of code: `to={`${this.props.event.key}/responsePage/${val.response_id}`}`
the application breaks on my end because like I said in another PR, the props aren't set when Im building and testing.

Because I am navigating to this component via the browser url `http://localhost:8080/responseList` there are no props.

**Comment by greeffjean** [comment]
> Hi Jean,
> 
> Great work! I really like what you did with the tag selection UI! Two suggestions to improve the UI a bit:
> 
> 1. I initially found the Update button on the tag selection dialog confusing, because its positioning and shape is the same as the tags, so I initially thought it was a selected tag or something. It would be clearer if the button was positioned differently or more prominent like you have it on the questions dialog
> 2. The reset button on the questions dialog resets both the questions and the tags - so it should probably be outside the question dialog, for example, on the same row as the Tags and Questions button, aligned to the right
> 
> Thanks!

Please see new UI updates, I think this new design is better

**Comment by greeffjean** [comment]
> Hey Jean, maybe I was imagining, but didn't you previously have the ability to click on the reponse_id and it takes you to the responsePage? That seems to have disappeared?

I found the issue, there was a missing event prop that was not being passed down

**Comment by avishkar58** [comment]
Updated UI looks great thanks!
You must have forgotten to commit some renames again, this time both the responseList.service and tagList.service, I pushed a quick fix though

**Review by avishkar58 — APPROVED** [review]

**Pull request merged**